crates/kornia-imgproc/src/features/fast.rs

```diff
@@ -1,162 +1,332 @@
+                    let lower_threshold = curr_pixel - self.threshold;
+                    let upper_threshold = curr_pixel + self.threshold;
+
+                    // Speed test: check 4 equidistant pixels first
+                    let mut speed_sum_b = 0;
+                    let mut speed_sum_d = 0;
+                    for &k in &[0, 4, 8, 12] {
+                        let ik = ((y as isize + ROW_OFFSETS[k]) * width as isize
+                            + (x as isize + COLUMN_OFFSETS[k]))
+                            as usize;
+                        let ring_pixel = src_slice[ik];
+                        if ring_pixel > upper_threshold {
+                            speed_sum_b += 1;
+                        } else if ring_pixel < lower_threshold {
+                            speed_sum_d += 1;
+                        }
+                    }
+
+                    if speed_sum_b < 3 && speed_sum_d < 3 {
+                        row[ix] = 0.0;
+                        continue;
+                    }
+
+                    // Full test for all 16 pixels
+                    for k in 0..16 {
+                        let ik = ((y as isize + ROW_OFFSETS[k]) * width as isize
+                            + (x as isize + COLUMN_OFFSETS[k]))
+                            as usize;
+                        let intensity = src_slice[ik];
+                        circle_intensities[k] = intensity;
+                        bins[k] = if intensity > upper_threshold {
+                            PIXEL_BRIGHTER
+                        } else if intensity < lower_threshold {
+                            PIXEL_DARKER
+                        } else {
+                            PIXEL_SIMILAR
+                        };
+                    }
+
+                    // Test for a contiguous arc of bright pixels
+                    let mut curr_response = corner_fast_response(
+                        curr_pixel,
+                        &circle_intensities,
+                        &bins,
+                        PIXEL_BRIGHTER,
+                        self.arc_length,
+                    );
+
+                    // If not found, test for a contiguous arc of dark pixels
+                    if curr_response == 0.0 {
+                        curr_response = corner_fast_response(
+                            curr_pixel,
+                            &circle_intensities,
+                            &bins,
+                            PIXEL_DARKER,
+                            self.arc_length,
+                        );
+                    }
+                    row[ix] = curr_response;
                 }
+            });
+
+        Ok(&self.corner_response)
+    }
+
+    /// Extracts keypoints using Non-Maximum Suppression on the corner response.
+    ///
+    /// # Returns
+    ///
+    /// Returns a vector of keypoint coordinates `[y, x]`.
+    pub fn extract_keypoints(&mut self) -> Vec<[usize; 2]> {
+        get_high_intensity_peaks(
```

Comment:
why not directly calling the function extract_keypoints ? i'd avoid having an overloaded method and use directly the function where is needed